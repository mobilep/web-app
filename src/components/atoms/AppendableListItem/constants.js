export const TYPES = {
	ADD: 'add',
	CHECK: 'check',
	STEP: 'step',
};

export default {
	TYPES,
};
