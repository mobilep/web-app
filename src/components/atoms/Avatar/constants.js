export const AVATAR_SIZES = {
	AVATAR_22: 'avatar-22',
	AVATAR_28: 'avatar-28',
	AVATAR_44: 'avatar-44',
	AVATAR_50: 'avatar-50',
	AVATAR_60: 'avatar-60',
	AVATAR_70: 'avatar-70',
	AVATAR_80: 'avatar-80',
	AVATAR_100: 'avatar-100',
	AVATAR_196: 'avatar-196',
};

export default {
	AVATAR_SIZES,
};
