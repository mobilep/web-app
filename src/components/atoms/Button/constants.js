export const COLORS = {
	DEFAULT: false,
	GREY: 'grey',
	RED: 'red',
	WHITE: 'white',
};
