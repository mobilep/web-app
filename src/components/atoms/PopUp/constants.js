const POSITIONS = {
	TOP_LEFT: 'topLeft',
	TOP_RIGHT: 'topRight',
	BOTTOM_RIGHT: 'bottomRight',
	BOTTOM_LEFT: 'bottomLeft',
	RIGHT: 'right',
	LEFT: 'left',
};

const BACKGROUND = {
	WHITE: 'white',
	RED: 'red',
};

export default {
	POSITIONS,
	BACKGROUND,
};
