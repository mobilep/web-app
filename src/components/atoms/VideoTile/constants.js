export const TILE_TYPES = {
	UPLOADING: 'uploading',
	ADD: 'add',
	DEFAULT: 'default',
};

export default {
	TILE_TYPES,
};
