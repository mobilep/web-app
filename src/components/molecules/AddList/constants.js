export const keyMap = {
	ENTER: 13,
	UP_ARROW: 38,
	DOWN_ARROW: 40,
	TAB: 9,
};

export default {
	keyMap,
};
