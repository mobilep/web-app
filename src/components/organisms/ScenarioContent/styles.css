.Scenarios-content {
	height: 100%;
	display: flex;
	flex-direction: column;
}

.Scenarios-body {
	overflow-y: auto;
	padding-right: 10px;
}

.Scenarios-sectionsBlock {
	display: grid;
	grid-template-areas:
		'steps'
		'bestPractice'
		'criterias'
		'dueDate'
		'people'
		'peopleDropdown';
	grid-template-columns: 100%;
	grid-template-rows: auto auto auto 1fr;
}

.Scenarios-steps {
	grid-area: steps;
	margin-right: 45px;
}

.Scenarios-criterias {
	grid-area: criterias;
	margin-right: 45px;
	margin-top: 90px;
}

.Scenarios-bestPractice {
	grid-area: bestPractice;
	margin-right: 45px;
	margin-top: 90px;
}

.Scenarios-people {
	grid-area: people;
	margin-left: 45px;
}

.Scenarios-peopleDropdown {
	grid-area: peopleDropdown;
	position: relative;
	margin-left: 45px;
	margin-top: 90px;
}

.Scenarios-dueDate {
	grid-area: dueDate;
}

.Scenarios-sectionExample {
	color: var(--grey3);
	font-size: 14px;
	padding-left: 36px;
}

@supports (display: grid) {
	.Scenarios-sectionsBlock > * {
		margin-left: 0;
		margin-right: 0;
		margin-top: 0;
	}
}

@media (min-width: 1500px) {
	.Scenarios-sectionsBlock {
		display: grid;
		grid-template-rows: auto auto auto 1fr;
	}
}

@media (max-width: 1500px) {
	.Scenarios-sectionsBlock {
		grid-template-rows: auto;
	}

	.Scenarios-people {
		margin-left: 0;
	}

	.Scenarios-sectionsBlock > div {
		width: 100%;
		max-width: unset;
	}
}

@media (max-width: 768px) {
	.Scenarios-content {
		overflow-y: unset;
	}

	.Scenarios-body {
		overflow-y: unset;
		padding-right: 0;
	}
}
