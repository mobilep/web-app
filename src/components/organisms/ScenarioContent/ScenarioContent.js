import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
	AppendableList,
	AppendableCriteriaList,
	ScenarioPeopleSection,
	ScenarioBestPracticeSection,
	ScenarioHeaderSection,
	ScenarioSection,
	ViewLoader,
	ScenarioDueDatePicker,
	ScenarioControls,
} from '../../';
import constants from '../../../constants';
import './styles.css';
import { LanguageContext } from '../../../utils';

const { common } = constants;
const { modes: { EDIT_MODE, CREATE_MODE } } = common;
const { Consumer: LanguageConsumer } = LanguageContext;

export default class ScenarioContent extends Component {

	static propTypes = {
		authorDetails: PropTypes.object,
		availableCriteriaItems: PropTypes.array,
		bestPracticeCards: PropTypes.array,
		canEditVideo: PropTypes.bool,
		content: PropTypes.object,
		criterias: PropTypes.array,
		currentScenarioContentFetching: PropTypes.bool,
		defaultDueDate: PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.instanceOf(Date)]),
		description: PropTypes.string,
		draft: PropTypes.bool,
		dueDate: PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.instanceOf(Date)]),
		fallBackVideoUrl: PropTypes.string,
		headerControllsHandlers: PropTypes.object,
		inboxData: PropTypes.array,
		mode: PropTypes.string,
		name: PropTypes.string,
		onBestPracticeRemove: PropTypes.func,
		onBestPracticeVideoAdd: PropTypes.func,
		onCriteriaCreate: PropTypes.func,
		onCriteriaRemove: PropTypes.func,
		onCriteriaSearch: PropTypes.func,
		onCriteriaSelect: PropTypes.func,
		onCurrentStepValueChange: PropTypes.func,
		onDescriptionChange: PropTypes.func,
		onDraftClick: PropTypes.func,
		onDueDateChange: PropTypes.func,
		onGroupSelect: PropTypes.func,
		onNameChange: PropTypes.func,
		onScenarioDeSelect: PropTypes.func,
		onStepAdd: PropTypes.func,
		onStepRemove: PropTypes.func,
		onUserRemove: PropTypes.func,
		onUserSearch: PropTypes.func,
		onUserSelect: PropTypes.func,
		onVideoChange: PropTypes.func,
		scenarioHeader: PropTypes.func,
		searchCriteriaItems: PropTypes.array,
		searchUserItems: PropTypes.array,
		selectedUsersCounter: PropTypes.number,
		showValidationCriteriasError: PropTypes.bool,
		showValidationDescriptionError: PropTypes.bool,
		showValidationNameError: PropTypes.bool,
		showValidationStepsError: PropTypes.bool,
		steps: PropTypes.array,
		teamData: PropTypes.object,
		type: PropTypes.string,
		updateDate: PropTypes.string,
		userItems: PropTypes.array,
		userSearchFetching: PropTypes.bool,
		validationCriteriasError: PropTypes.string,
		validationDescriptionError: PropTypes.string,
		validationNameError: PropTypes.string,
		validationStepsError: PropTypes.string,
		videoId: PropTypes.string,
		videoState: PropTypes.string,
		videoThumbnailUrl: PropTypes.string,
		videoUrl: PropTypes.string,
	}

	isContentEditable ({ mode } = this.props) {
		return mode === EDIT_MODE || mode === CREATE_MODE;
	}

	shouldRenderBestPracticeSection ({ bestPracticeCards = [] } = this.props) {
		return this.isContentEditable() || !!bestPracticeCards.length;
	}

	render () {
		const {
			authorDetails,
			availableCriteriaItems,
			bestPracticeCards,
			content,
			criterias,
			currentScenarioContentFetching,
			description,
			draft,
			fallBackVideoUrl,
			headerControllsHandlers,
			inboxData,
			mode,
			name,
			onBestPracticeRemove,
			onBestPracticeVideoAdd,
			onCriteriaCreate,
			onCriteriaRemove,
			onCriteriaSearch,
			onCriteriaSelect,
			onCurrentStepValueChange,
			onDueDateChange,
			onDescriptionChange,
			onDraftClick,
			onGroupSelect,
			onNameChange,
			onScenarioDeSelect,
			onStepAdd,
			onStepRemove,
			onUserRemove,
			onUserSearch,
			onUserSelect,
			onVideoChange,
			searchUserItems,
			selectedUsersCounter,
			showValidationCriteriasError,
			showValidationDescriptionError,
			showValidationNameError,
			showValidationStepsError,
			steps,
			teamData = {},
			type,
			updateDate,
			userItems,
			userSearchFetching,
			validationCriteriasError,
			validationDescriptionError,
			validationNameError,
			validationStepsError,
			videoId,
			videoState,
			videoThumbnailUrl,
			videoUrl,
			dueDate,
			defaultDueDate,
			canEditVideo,
		} = this.props;
		if (currentScenarioContentFetching) return <ViewLoader />;
		const isEditable = this.isContentEditable();

		return (
			<LanguageConsumer>
				{({ common, scenarios, _contextSettings: { dateFnsLanguageCode } }) => (
					<div className='Scenarios-content'>
						<ScenarioControls
							{...content.headerSection[mode]}
							{...headerControllsHandlers}
							onDraftClick={onDraftClick}
							mode={mode}
							draft={draft}
							scenarioType={type}
							updateDate={updateDate}
							canEditVideo={canEditVideo}
						/>

						<div className='Scenarios-body'>
							<ScenarioHeaderSection
								addVideoText={common.addVideo}
								changeVideoText={common.changeVideo}
								authorDetails={authorDetails || {}}
								showValidationNameError={showValidationNameError}
								validationNameError={validationNameError}
								scenarioType={type}
								showValidationDescriptionError={showValidationDescriptionError}
								validationDescriptionError={validationDescriptionError}
								mode={mode}
								canEditVideo={canEditVideo}
								count={selectedUsersCounter}
								header={name}
								description={description}
								thumbnailUrl={videoThumbnailUrl}
								videoState={videoState}
								videoUrl={videoUrl}
								videoId={videoId}
								fallBackVideoUrl={fallBackVideoUrl}
								onHeaderChange={onNameChange}
								onDescriptionChange={onDescriptionChange}
								onScenarioDeSelect={onScenarioDeSelect}
								onVideoChange={onVideoChange}
								{...content.headerSection[mode]}
								{...headerControllsHandlers}
								mobileHeader={scenarios.scenario}
								draft={draft}
								saveDraftText={scenarios.saveDraftButton}
								onDraftClick={onDraftClick}
								updateDate={updateDate}
							/>
							<div className='Scenarios-sectionsBlock'>
								{/* STEPS */}
								<ScenarioSection
									showValidationError={showValidationStepsError}
									errorText={validationStepsError}
									className='Scenarios-steps'
									header={content.steps}>
									<AppendableList
										editable={isEditable}
										placeholder={content.addSteps}
										onChangeItem={onCurrentStepValueChange}
										items={steps}
										onAddItem={onStepAdd}
										type={'step'}
										onRemoveItem={onStepRemove}
									/>
									{isEditable && !steps.length &&
										<div className='Scenarios-sectionExample'>{scenarios.stepExample}</div>
									}
								</ScenarioSection>

								{/* PEOPLE */}
								<ScenarioSection
									className='Scenarios-people'
									header={content.peopleSection.peoplePracticing} counter={selectedUsersCounter}>
									<ScenarioPeopleSection
										userSearchFetching={userSearchFetching}
										editable={isEditable}
										userItems={userItems}
										searchUserItems={searchUserItems}
										onUserSelect={onUserSelect}
										onGroupSelect={onGroupSelect}
										onUserRemove={onUserRemove}
										onUserSearch={onUserSearch}
										groupList={teamData.teamList || []}
										inboxData={inboxData}
									/>
								</ScenarioSection>
								<div className='Scenarios-peopleDropdown' />

								{/* CRITERIA */}
								<ScenarioSection
									showValidationError={showValidationCriteriasError}
									errorText={validationCriteriasError}
									className='Scenarios-criterias'
									header={content.evaluationCriteria}>
									<AppendableCriteriaList
										editable={isEditable}
										content={content.criteriaSection}
										criteriaItems={criterias}
										onCriteriaRemove={onCriteriaRemove}
										onCriteriaCreate={onCriteriaCreate}
										availableCriteriaItems={availableCriteriaItems}
										onCriteriaSelect={onCriteriaSelect}
										onCriteriaSearch={onCriteriaSearch}
									/>
									{isEditable && !criterias.length &&
										<div className='Scenarios-sectionExample'>
											{scenarios.evaluationCriteriaExample}
										</div>
									}
								</ScenarioSection>

								{/* BEST PRACTICES */}
								{this.shouldRenderBestPracticeSection() &&
									<ScenarioBestPracticeSection
										{...{ [mode]: true }}
										onVideoAdd={onBestPracticeVideoAdd}
										onVideoRemove={onBestPracticeRemove}
										cards={bestPracticeCards}
										className='Scenarios-bestPractice'
									/>}

								{/* DUE DATE */}
								{(dueDate || isEditable) &&
									<ScenarioSection
										className='Scenarios-dueDate'
										header={content.dueDate}
									>
										<ScenarioDueDatePicker
											selectedDate={dueDate || defaultDueDate}
											editable={isEditable}
											locale={dateFnsLanguageCode}
											onDateChange={onDueDateChange}
										/>
									</ScenarioSection>
								}
							</div>
						</div>
					</div>
				)}
			</LanguageConsumer>
		);
	}

}
