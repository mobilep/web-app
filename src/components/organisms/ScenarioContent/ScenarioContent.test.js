import React from 'react';
import { shallow } from 'enzyme';
import ScenarioContent from './ScenarioContent';
import constants from '../../../constants';
const { content } = constants;

describe('Organism - ScenarioContent', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<ScenarioContent content={content.en.scenarios} />
		);
		expect(wrapper).toMatchSnapshot();
	});

});
