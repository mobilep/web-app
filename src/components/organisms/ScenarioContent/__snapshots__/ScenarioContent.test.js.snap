// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organism - ScenarioContent Should render 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
